### ISSUE TITLE
Cosine update grad is inf in some blocks, it's ok?

### ISSUE THREAD

**Issue by germanjke** [mosaicml/llm-foundry#282]
I'm observe my optimizer metrics while mpt trains, and some blocks are infs e.g.

`Train cosine/update_grad/model._fsdp_wrapped_module.transformer.blocks.9._fsdp_wrapped_module.ffn.down_proj.weight: inf`

It's ok or why this happens? Do you know this issue?

**Comment by vchiley** [comment]
That seems weird / should't be happening.
Is the model still training ie loss still going down?

**Comment by vchiley** [comment]
Is this issue related to [this](https://github.com/mosaicml/llm-foundry/issues/284)?
Are they effectively duplicates? Should we close this issue?

**Comment by germanjke** [comment]
> That seems weird / should't be happening. Is the model still training ie loss still going down?

yes still training some time, but after something like 5-10 k steps is explodes

**Comment by vchiley** [comment]
ok that makes sense.
The precursor to loss spikes is poorly behaved weights.
Closing issue. See discussion [here](https://github.com/mosaicml/llm-foundry/issues/284#issuecomment-1583616361).
